### PR TITLE
osd/PGStateUtils: initialize NamedState::enter_time.

### PR DESCRIPTION
**Pull request by majianpeng**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/osd/PGStateUtils.cc [hunk 1]
```diff
@@ -10,6 +10,7 @@ NamedState::NamedState(PGStateHistory *pgsh, const char *state_name)
   if(pgsh) {
     pgsh->enter(ceph_clock_now(), state_name);
   }
+  enter_time = ceph_clock_now();
```
Comment (tchaikov) [inline src/osd/PGStateUtils.cc]: could you instead initialize `enter_time` with initialize list, and use it for updating `pgsh` to be consistent ? like
```c++
NamedState::NamedState(PGStateHistory *pgsh, const char *state_name)
  : pgsh(pgsh), state_name(state_name), enter_time(ceph_clock_now()) {
  if (pgsh) {
    pgsh->enter(enter_time, state_name);
  }
}
```

also might want to note, this is a regression introduce by d96c9072b72c7e8cf3757dc3f28afeebdce1cf66

**Comment by tchaikov** [comment]
@majianpeng since octopus has been forked from master. i 

- rebased your change onto octopus instead, and 
- changed the title of "osd/PGStatueUtils: remove unused member variable" to note that it's the member variable which gets removed, not a parameter.

**Comment by tchaikov** [comment]
@majianpeng could you please address the https://github.com/ceph/ceph/pull/33813#issuecomment-596900222 ? i think you reverted my change.

**Comment by majianpeng** [comment]
@tchaikov . i updated comments by your suggestions. But  how to "rebased your change onto octopus instead"?

**Comment by tchaikov** [comment]
> @tchaikov . i updated comments by your suggestions. But how to "rebased your change onto octopus instead"?

octopus is the next release branch. you could fetch it and use `git cherry-pick` to cherry-pick your change in that branch.

@majianpeng i am pushing to your branch again. please avoid rewriting it.

**Comment by liewegas** [comment]
retest this please

**Comment by majianpeng** [comment]
@tchaikov . i push one  branch using force ,it also update this branch. But i didn't change anything.

**Comment by liewegas** [comment]
@majianpeng please rebase on ``octopus`` instead of ``master``...

**Comment by tchaikov** [comment]
> @tchaikov . i push one branch using force ,it also update this branch. But i didn't change anything.

i think this PR was good. please don't overwrite it again.

**Pull request merged**